HarpGuru/HarpGuru/components/index.tsx

```diff
@@ -0,0 +1,96 @@
+import 'react-native-gesture-handler'
+
+import {PanGestureHandler, PanGestureHandlerGestureEvent, State} from 'react-native-gesture-handler'
+import {StyleSheet, View} from 'react-native'
+import React, { useState, useRef, useEffect } from 'react'
+import type { ReactElement } from 'react'
+import { getApparatusIds, getPozitionIds, getPitchIds, getHarpStrata } from 'harpstrata'
+import type { ActiveIds, HarpStrata, HarpStrataProps } from 'harpstrata'
+
+import { themeSizes } from '../../Styles'
+import { HomeScreen } from '../../Screens'
+import {AnimatedMenuContainer, CovariantMenu, LayoutMenu} from '../../OverlayMenu'
+import {DisplayModes} from '../../HarpFace'
+
+const styles = StyleSheet.create({
+  overlay: {
+    ...StyleSheet.absoluteFillObject,
+  }
+})
+
+const { 8: swipeThreshold } = themeSizes
+
+const [ initialApparatusId ] = getApparatusIds()
+const [ initialPozitionId ] = getPozitionIds()
+const [ initialPitchId ] = getPitchIds()
+const initialActiveIds: ActiveIds = []
+
+const initialHarpStrataProps: HarpStrataProps = {
+  apparatusId: initialApparatusId,
+  pozitionId: initialPozitionId,
+  harpKeyId: initialPitchId,
+  activeIds: initialActiveIds,
+}
+const initialHarpStrata: HarpStrata = getHarpStrata(initialHarpStrataProps)
+const { Degree: initialDisplayMode } = DisplayModes
+
+const usePrevious = (value: State) => {
```

Comment:
This doesn't actually seem to be the case. In fact I expect that I can provide a single helper for every occasion which I need usePrevious and it will somehow recognise each instance which has called it independently.

This probably has something to do with the way hooks work in general. I need to learn more.

This can be demonstrated by adding the following to the usePrevious function:

```
const usePrevious = (value: State) => {
  const ref = useRef(Math.ceil(Math.random() * 1000) + '-')
  console.log(':::::::::::::::::::::::::::::::::: ' + ref.current)
  useEffect(() => {
    const [ prefix ] = ref.current.split && ref.current.split('-')
    ref.current = prefix + '-' + value
  })
  return ref.current
}
```
This will assign a random prefix to each of the different instances ref.current values which will persist through all of that objects uses of this hook. Using this we can see that the previous value of one instance does not impact the previous value of another.

Reply:
Done. A helper folder has been created with a `usePrevious` export.